### PR TITLE
d0796r1: Fix Listing 1 to have correct syntax

### PR DESCRIPTION
**Pull request by mhoemmen**
Listing 1 in d0796r1 had incorrect syntax.
Fix the example so readers won't object.
Reword surrounding text for correctness and clarity.

### CODE REVIEW

File: affinity/cpp-20/d0796r1.md [hunk 1]
```diff
@@ -37,28 +37,27 @@ The affinity problem is especially challenging for applications whose behavior c
...
+  [=] (double& a_i) { scalar * a_i; });
+
+// Use future affinity interface to migrate data at next
+// use and move pages closer to next accessing thread.
 ...
-//now faster, because data is local now
+// Faster, because data are local now.
 std::for_each(par, std::begin(a), std::end(a),
-[=](double b, double c){b[i]+scalar*c[i]});
+  [=] (double& a) { scalar * a_i; } );
```
[the remark this answers is not in the document]
Comment (Ruyk) [reply affinity/cpp-20/d0796r1.md]: shouldn't this be:
```
 std::for_each(std::begin(v), std::end(v), 
                [=] (double& a_i) { a_i *= scalar; });
```
?

[the remark this answers is not in the document]
Comment (mhoemmen) [reply affinity/cpp-20/d0796r1.md]: @Ruyk Both correct; thanks!

[the remark this answers is not in the document]
Comment (mhoemmen) [reply affinity/cpp-20/d0796r1.md]: @Ruyk Should I resubmit or would you like to make the changes?

[the remark this answers is not in the document]
Comment (Ruyk) [reply affinity/cpp-20/d0796r1.md]: Just did it with the online editor

**Pull request merged**